packages/ui/src/app/pages/Settings/Settings.tsx

```diff
@@ -65,6 +99,68 @@ export const Settings = () => {
                     onChange={switchNetwork}
                     selectSize="l"
                   />
+                  {network == 'custom' && (
+                    <>
+                      <SettingsWarningInformation icon={<WarnedIcon />} title="Attention">
+                        <ColumnGapBlock gap={5}>
+                          <TextMedium lighter>
+                            Please beware that using 3rd party QN may result in fraud. Only use trusted providers
+                          </TextMedium>
+                        </ColumnGapBlock>
+                      </SettingsWarningInformation>
+                      <InputComponent
+                        label={t('customFaucet')}
+                        validation={/^(http|https):\/\//i.test(customFaucetEndpoint) ? undefined : 'invalid'}
```

Comment:
[The design](https://www.figma.com/file/ZwWMTt8ZvEPmFekDHCBWe1/Settings?type=design&node-id=4-8214&mode=design&t=5XjKn64QWU2WMMVo-4) implies that the validation should check if the endpoints are reachable:
![image](https://github.com/Joystream/pioneer/assets/6571453/aeb478fa-51fe-41e9-9463-b8cf9fa5b139)

The rest of the app is now using [React Hook Form](https://react-hook-form.com/) I think it should be used here too.

Reply:
Saying about checking connection, When will be the best point to do it? I guess while the user enters the url, it could be a bit complicated to check connection,  also I thought it could be not good point when the user click "save setting" button for checking connection. so i just implemented the validation check for url format..

Reply:
The url validation is good we should keep this. But currently if someone makes a mistake when entering one of the endpoint they might not realize it for a while. So I think we should somehow check the 3 endpoints:
- For the `rpc` not it would be just opening a websocket and has soon as it's opened closing it.
- For the QN there should be a quick way to check that this is a valid gql server :crossed_fingers: .
- For the faucet there is a status endpoint at `../status` (so for `faucet.joystream.org/member-faucet/register` it's `faucet.joystream.org/member-faucet/status`).

You're right about avoiding validating while the users are typing. I think the validation should happen on input blur but after clicking on the button we should show a loader until the validation is completed (then it either update the local storage and reload the page if all endpoints were reached or show the "Connection error" if not).

Reply:
@thesan I have updated the Endpoint validation code.